### PR TITLE
Update Debian CI to Bookworm, Ubuntu CI to Noble

### PR DESCRIPTION
**Pull request by janisozaur**
Apparently I've published the image 3 months ago (https://github.com/OpenRCT2/openrct2-docker-build/commit/cecfe56ea3319b434f31827ae16b86afc986bef4), but never updated this repo.

### CODE REVIEW

**Comment by janisozaur** [comment]
I think I know why it wasn't merged. Bookworm uses the buggy GCC 12 release, which fails to compile OpenRCT2

**Comment by janisozaur** [comment]
I had to drop all the optimizations to avoid GCC bugs. That makes it unworthy for release, but perhaps we could try coming up with exclusions for just the offending files, there are only 4 of them:
- src/openrct2/localisation/FormatCodes.cpp
- src/openrct2/platform/Platform.Posix.cpp
- src/openrct2/ride/Ride.cpp
- src/openrct2/scenario/Scenario.cpp
(https://github.com/OpenRCT2/OpenRCT2/actions/runs/10256004755/job/28374225995)

**Comment by ZehMatt** [comment]
Can we perhaps jump to GCC 13?

**Comment by janisozaur** [comment]
Not without too much hassle - and we'd likely make OpenRCT2 not compatible with this Debian release. I prefer dealing with it the way it's done currently

**Comment by ZehMatt** [comment]
How about GCC 12.1 or even newer? Having to compensate for compiler bugs is quite annoying.

**Comment by janisozaur** [comment]
The bugs are present throughout GCC12 series. Bookworm actually uses 12.2

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -305,19 +305,14 @@ jobs:
...
-            image: openrct2/openrct2-build:14-bullseye
-            build_flags: -DCMAKE_POSITION_INDEPENDENT_CODE=on -DCMAKE_CXX_FLAGS="-g -gz -fno-var-tracking-assignments" -DWITH_TESTS=off
-          - platform: i686
-            distro: Ubuntu
-            release: jammy
-            image: openrct2/openrct2-build:14-jammy32
-            build_flags: -DFORCE32=ON -DENABLE_SCRIPTING=OFF -DCMAKE_CXX_FLAGS="-m32 -g -gz" -DWITH_TESTS=off
+            release: bookworm
+            image: openrct2/openrct2-build:15-bookworm
+            build_flags: -DCMAKE_POSITION_INDEPENDENT_CODE=on -DCMAKE_CXX_FLAGS="-g -gz -fno-var-tracking-assignments" -DWITH_TESTS=off -DCMAKE_BUILD_TYPE=Debug # overwrite the default build type to avoid GCC 12 bugs
```
Comment (janisozaur) [inline .github/workflows/ci.yml:315]:
```suggestion
            build_flags: -DCMAKE_POSITION_INDEPENDENT_CODE=on -DCMAKE_CXX_FLAGS="-g -gz -fno-var-tracking-assignments" -DWITH_TESTS=off
```

**Comment by Gymnasiast** [comment]
Would it be possible to keep the 32-bit Jammy for the users that still need it? I understand it would be less optimised due to GCC 12, but it is our only 32-bit Linux image left.

If it’s too much hassle, I would be fine with dropping it.

**Comment by AaronVanGeffen** [comment]
It's a pity the 32-bit packages are missing on modern Ubuntu 64-bit distros, but I would prefer to move away from Jammy entirely, tbh. 

Do we know if there are any people still using 32-bit versions at all? I think we could easily drop 32-bit Windows builds as well — but that's another discussion.

**Comment by janisozaur** [comment]
No, @AaronVanGeffen 's PR caused hangs on jammy i686, which is still on GCC 11, not 12. Maybe I can launch that on bookworm instead, I can try it out later on.

**Review by Gymnasiast — APPROVED** [review]

**Comment by janisozaur** [comment]
There is libzip available for i686 in bookworm https://packages.debian.org/bookworm/libzip-dev

**Review by AaronVanGeffen — APPROVED** [review]

**Comment by cheweytoo** [comment]
Just FYI: This breaks running the AppImage build on Ubuntu 22.04.
Which I guess is unavoidable without quite a bit of additional effort.

```
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.31' not found (required by ./OpenRCT2-x86_64.AppImage)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by ./OpenRCT2-x86_64.AppImage)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by ./OpenRCT2-x86_64.AppImage)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by ./OpenRCT2-x86_64.AppImage)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libcrypto.so.3)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.36' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libzip.so.4)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libicuuc.so.74)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libicuuc.so.74)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libSDL2-2.0.so.0)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libSDL2-2.0.so.0)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libgssapi_krb5.so.2)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libldap.so.2)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/liblber.so.2)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libXcursor.so.1)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libwayland-client.so.0)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libxkbcommon.so.0)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libdecor-0.so.0)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libgnutls.so.30)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libkrb5.so.3)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libk5crypto.so.3)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libkrb5support.so.0)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libsasl2.so.2)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libpulsecommon-16.1.so)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libdbus-1.so.3)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libp11-kit.so.0)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libtasn1.so.6)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libsndfile.so.1)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libsndfile.so.1)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libsystemd.so.0)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libapparmor.so.1)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libmp3lame.so.0)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libcap.so.2)
./OpenRCT2-x86_64.AppImage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_OpenRCR6axhH/usr/bin/../lib/libgcrypt.so.20)
```

**Pull request merged**